src/briefcase/commands/new.py

```diff
@@ -516,16 +646,154 @@ def verify_tools(self):
+        :param target: The path where the newly created pyproject.toml file should be migrated to.
+        """
+        self.merge_or_copy_pyproject(project_dir, target)
+        license_file = target / "LICENSE"
+        if not license_file.is_file():
+            self.logger.warning(
+                f"License file not found in {target}. You should either create a new license file in {target}"
+                " or rename an already existing license file to LICENSE."
+            )
+
+        changelog_file = target / "CHANGELOG"
+        if not changelog_file.is_file():
+            self.logger.warning(
+                f"Changelog file not found in {target}. You should either create a new changelog file in"
+                f" {target} or rename an already existing changelog file to CHANGELOG."
+            )
+
+    @contextmanager
+    def use_tmp_base_path(self):
+        """Contextmanager to use a temporary directory as base path."""
+        try:
+            tmpdir = TemporaryDirectory()
+            base_path = self.base_path
+            self.base_path = Path(tmpdir.name)
+            yield self.base_path
+        finally:
+            self.base_path = base_path
+            tmpdir.cleanup()
+
+    def setup_existing_app_for_briefcase(
+        self,
+        template: Optional[str] = None,
+        template_branch: Optional[str] = None,
+        target: Path = None,
+        **options,
+    ):
+        """Run the wizard in a temporary directory and copy the necessary files.
+
+        :param template: The cookiecutter template to use.
+        :param template_branch: The git branch that the template should use.
+        :param target: The path to the project that should be set up for briefcase.
+        """
+        # Cannot setup new app if it already has briefcase settings in pyproject
+        if (target / "pyproject.toml").exists():
+            with open(target / "pyproject.toml", "rb") as file:
+                pyproject = tomllib.load(file)
+            if "tool" in pyproject and "briefcase" in pyproject["tool"]:
+                raise ValueError(
+                    f"[tool.briefcase] already in {target / 'pyproject.toml'}."
+                    " Cannot initialise briefcase for an application that already is packaged with briefcase."
+                )
+
+        with self.use_tmp_base_path() as output_path:
+            app_name = self.new_app(
+                template=template,
+                template_branch=template_branch,
+                target=target,
+                **options,
+            )
+            project_dir = output_path / app_name
+
+            self.migrate_necessary_files(project_dir, target)
+
+    def create_and_move_new_app(
+        self,
+        template: Optional[str] = None,
+        template_branch: Optional[str] = None,
+        target: Path = None,
+        **options,
+    ):
+        """Create a new briefcase app in a given directory.
```

Comment:
I'm not sure I follow why creating in a temp folder and copying is needed for the "new app" case. There doesn't appear to be any re-use of the "generate into temp folder" logic (i.e., always generate into a template folder, but varying what is copied out) - there's two independent uses of the `tmp_base_path` context manager.

If the issue is creating into a specific folder - wouldn't that be better served by making the base path configureable when the template is rolled out?

Reply:
The reason we're creating a temp folder is that currently, `--target` is used to specify the exact directory where Briefcase should be set up, and the cookiecutter template (as far as we understand) creates a new directory with the name `app_name`.

So `__call__` essentially has three options right now:

 * No `--target`: no change to behaviour
 * `--target` exists: Briefcase assumes that `--target` points to a python project and attempts to set it up for briefcase
 * `--target` doesn't exist, but the parent of `--target` does exist: Briefcase runs the wizard and creates a new project with root directory `--target` (i.e. not `--target/app_name`)

We could remove the last option altogether and only accept existing targets, but check if there's a `setup.py`/`setup.cfg`/`pyproject.toml` file in the target. Then, if there is no setup-file, Briefcase could create a new directory `--target/app_name` where the project is initialised. In that case, we wouldn't need the temporary directory when creating new Briefcase apps.

What do you think?

Reply:
I don't think this is quite right - or, at least, it's potentially skipping over a decision that we should make explicitly.

To my eyes, there's 4 possible cases at runtime:
1. No `--target`
2. `--target` points a non-existent directory
3. `--target` points at an existing directory that does *not* contain `setup.py/setup.cfg/pyproject.toml`
4. `--target` points at an existing directory that contains `setup.py/setup.cfg/pyproject.toml`

The first and last cases are the easy ones. Case 1 is existing behavior; the wizard asks for the new project name, and creates a directory with that name. Case 4 is the "migrate existing" case; the wizard reads whatever it can from the metadata in the target folder, and generates pyproject.toml accordingly.

The question is what to do with the remaining 2 cases. I can see two possibilities, depending on whether we consider `--target` to be a "migration" option or a "destination" option.

If it's a "migration" option, case 2 and 3 are errors. If you specify a target that doesn't contain project metadata, it's not an existing project, so we should error out and let the user know that the specified target isn't an existing project.

If it's a "destination" option, then we're specifying where we want the final product to end up. In that case, it doesn't matter whether the directory exists or not; what we're specifying is where we want the final files to end up. If there's metadata there, that's just another source of wizard hints (which is case 4); but if there isn't metadata, the last name part of the target folder becomes the app name, and the final `pyproject.toml` is generated in whatever target was specified. If the folder *didn't* exist, then we can copy in all the template code; if it did, we just create the pyproject.toml.

I guess there's also middle ground: Make case 1 and case 2 the same, except that in case 1, you have to ask the user for the app name; and make case 3 and 4 the same, except that case 3 needs to take the app name from the directory, rather than app metadata. 

Based on your explanation, I *think* you're leaning towards the "migration" definition - and I think I agree. The "destination" interpretation is going to be a more complicated to implement, and will be prone to randomly creating directories full of code because someone mistyped a directory name - and at the end of the day, it's a lot of complexity for a use case that I'm not sure really exists. You're either starting a new project, or you're converting an existing one, and for all the complications of cases 2 and 3, the alternate implementation is `mv oldname newname`. :-)
 
My only hesitation with this comes from the naming of `--target`. The name `target` makes it sound like it's a "destination" argument - which it definitely isn't under the "migration" interpretation. I'm tempted to suggest that we should replace ``--target`` with a simple boolean ``--convert`` option, and *only* support converting the current working directory. I'm not sure there's much of a use case for "migrate that project in that directory over there"; "make my current project Briefcase compatible" is a lot more likely. 

The other option, of course, would be to have a different command entirely - possibly sharing a lot of common implementation, but `briefcase convert` converts a pyproject.toml in your current folder, and `briefcase new` creates an entirely new project directory.

Thoughts?

Reply:
That's a good point! We agree that the boolean `--convert` option might be easier to use and more maintainable than the `--target` option, and the other option of having a `briefcase convert` command sounds even better. 

Having a `briefcase convert` command would probably be more discoverable for new users -- it'd be immediately visible by just writing `briefcase` in the terminal. This option would also maybe be easier to maintain since we would more clearly separate the logic for converting an app for `briefcase` and setting up a new app for briefcase.